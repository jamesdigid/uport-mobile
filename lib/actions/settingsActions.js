// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import { 
  SET_CHANNEL,
  START_SWITCHING_DATABACKUP,
  SET_DATABACKUP,
  SET_OPT_OUT,
  SET_DATA_CUSTOM_URL } from '../constants/SettingsActionTypes'

export function setChannel (channel) {
  return {
    type: SET_CHANNEL,
    channel
  }
}

export function startSwitchingDataBackup (isOn) {
  return {
    type: START_SWITCHING_DATABACKUP,
    isOn
  }
}

export function setDataCustomUrl (customUrl) {
  return {
    type: SET_DATA_CUSTOM_URL,
    customUrl
  }
}

export function setDataBackup (isOn) {
  return {
    type: SET_DATABACKUP,
    isOn,
    _backup: true
  }
}

export function setAnalyticsOptOut (analyticsOptOut) {
  return {
    type: SET_OPT_OUT,
    analyticsOptOut
  }
}
