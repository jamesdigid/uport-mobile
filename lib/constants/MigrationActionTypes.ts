//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//

export const ADD_MIGRATION_TARGET = 'ADD_MIGRATION_TARGET'
export const RUN_MIGRATIONS = 'RUN_MIGRATIONS'
export const RUN_MIGRATION_STEP = 'RUN_MIGRATION_STEP'
export const STARTED_MIGRATION_STEP = 'STARTED_MIGRATION_STEP'
export const COMPLETED_MIGRATION_STEP = 'COMPLETED_MIGRATION_STEP'
export const FAILED_MIGRATION_STEP = 'FAILED_MIGRATION_STEP'


export enum MigrationTarget {
  PreHD = "PreHD"
}

export enum MigrationStep {
  IdentityManagerChangeOwner = 'IdentityManagerChangeOwner',
  UpdatePreHDRootToHD = 'UpdatePreHDRootToHD',
  UportRegistryDDORefresh = 'UportRegistryDDORefresh'
}

export enum MigrationStatus {
  NotStarted,
  Started,
  Completed,
  Error
}

export interface TargetAction {
  type: 'ADD_MIGRATION_TARGET' | 'RUN_MIGRATIONS',
  target: MigrationTarget
}

export interface StepAction {
  type: 'RUN_MIGRATION_STEP' | 'STARTED_MIGRATION_STEP' | 'COMPLETED_MIGRATION_STEP' | 'FAILED_MIGRATION_STEP',
  step: MigrationStep
}