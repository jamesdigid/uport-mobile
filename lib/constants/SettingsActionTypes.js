// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
export const SET_CHANNEL = 'SET_CHANNEL'
export const SET_DATABACKUP = 'SET_DATABACKUP'
export const SET_DATA_CUSTOM_URL = 'SET_DATA_CUSTOM_URL'
export const START_SWITCHING_DATABACKUP = 'START_SWITCHING_DATABACKUP'
export const SET_OPT_OUT = 'SET_OPT_OUT'
