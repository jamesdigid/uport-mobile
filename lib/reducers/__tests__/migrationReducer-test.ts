// This file is part of the uPort Mobile App.

// The uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.

// The uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.

// You should have received a copy of the GNU General Public License
// along with The uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.

// (C) Copyright 2016-2018 ConsenSys AG

import { addMigrationTarget, startedMigrationStep, completedMigrationStep, failedMigrationStep } from 'uPortMobile/lib/actions/migrationActions'
import { MigrationStep, MigrationTarget, MigrationStatus } from 'uPortMobile/lib/constants/MigrationActionTypes'
import reducer, { MigrationState } from '../migrationsReducer'


const initialState : MigrationState = {
  targets: [],
  steps: {}
}

const populated : MigrationState = {
  targets: [MigrationTarget.PreHD],
  steps: {}
}
populated.steps[MigrationStep[MigrationStep.IdentityManagerChangeOwner]] = MigrationStatus.Started

describe('addMigrationTarget', () => {
  it('adds a target', () => {
    expect(reducer(initialState, addMigrationTarget(MigrationTarget.PreHD))).toMatchSnapshot()
  })

  it('does not add an existing target', () => {
    expect(reducer(populated, addMigrationTarget(MigrationTarget.PreHD))).toEqual(populated)
  })
})

describe('startedMigrationStep', () => {
  it('starts a step', () => {
    expect(reducer(initialState, startedMigrationStep(MigrationStep.IdentityManagerChangeOwner))).toMatchSnapshot()
  })

  it('does not a step if already started', () => {
    expect(reducer(populated, startedMigrationStep(MigrationStep.IdentityManagerChangeOwner))).toEqual(populated)
  })

  it('should start an additional step', () => {
    expect(reducer(populated, startedMigrationStep(MigrationStep.UpdatePreHDRootToHD))).toMatchSnapshot()
  })
})

describe('completedMigrationStep', () => {
  it('completes a step', () => {
    expect(reducer(initialState, completedMigrationStep(MigrationStep.IdentityManagerChangeOwner))).toMatchSnapshot()
  })

  it('completes a step if already started', () => {
    expect(reducer(populated, completedMigrationStep(MigrationStep.IdentityManagerChangeOwner)).steps[MigrationStep.IdentityManagerChangeOwner]).toEqual(MigrationStatus.Completed)
  })

  it('should complete an additional step', () => {
    expect(reducer(populated, completedMigrationStep(MigrationStep.UpdatePreHDRootToHD))).toMatchSnapshot()
  })
})

describe('failedMigrationStep', () => {
  it('fails a step', () => {
    expect(reducer(initialState, failedMigrationStep(MigrationStep.IdentityManagerChangeOwner))).toMatchSnapshot()
  })

  it('fails a step if already started', () => {
    expect(reducer(populated, failedMigrationStep(MigrationStep.IdentityManagerChangeOwner)).steps[MigrationStep.IdentityManagerChangeOwner]).toEqual(MigrationStatus.Error)
  })

  it('should fail an additional step', () => {
    expect(reducer(populated, failedMigrationStep(MigrationStep.UpdatePreHDRootToHD))).toMatchSnapshot()
  })
})
