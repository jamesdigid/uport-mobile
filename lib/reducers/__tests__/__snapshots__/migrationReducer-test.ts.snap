// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addMigrationTarget adds a target 1`] = `
Object {
  "steps": Object {},
  "targets": Array [
    "PreHD",
  ],
}
`;

exports[`completedMigrationStep completes a step 1`] = `
Object {
  "steps": Object {
    "IdentityManagerChangeOwner": 2,
  },
  "targets": Array [],
}
`;

exports[`completedMigrationStep should complete an additional step 1`] = `
Object {
  "steps": Object {
    "IdentityManagerChangeOwner": 1,
    "UpdatePreHDRootToHD": 2,
  },
  "targets": Array [
    "PreHD",
  ],
}
`;

exports[`failedMigrationStep fails a step 1`] = `
Object {
  "steps": Object {
    "IdentityManagerChangeOwner": 3,
  },
  "targets": Array [],
}
`;

exports[`failedMigrationStep should fail an additional step 1`] = `
Object {
  "steps": Object {
    "IdentityManagerChangeOwner": 1,
    "UpdatePreHDRootToHD": 3,
  },
  "targets": Array [
    "PreHD",
  ],
}
`;

exports[`startedMigrationStep should start an additional step 1`] = `
Object {
  "steps": Object {
    "IdentityManagerChangeOwner": 1,
    "UpdatePreHDRootToHD": 1,
  },
  "targets": Array [
    "PreHD",
  ],
}
`;

exports[`startedMigrationStep starts a step 1`] = `
Object {
  "steps": Object {
    "IdentityManagerChangeOwner": 1,
  },
  "targets": Array [],
}
`;
